pallets/file-system/src/utils.rs

```diff
@@ -1208,49 +1177,80 @@ where
         // Return the threshold needed for the BSP to be able to volunteer for the storage request.
         T::HashToThresholdType::convert(volunteering_hash)
     }
-}
-
-impl<T: crate::Config> shp_traits::SubscribeProvidersInterface for Pallet<T> {
-    type ProviderId = ProviderIdFor<T>;
 
-    fn subscribe_bsp_sign_off(_who: &Self::ProviderId) -> DispatchResult {
-        // Adjust bsp assignment threshold by applying the inverse of the decay function after removing the asymptote
-        let mut bsp_assignment_threshold = BspsAssignmentThreshold::<T>::get();
-        let base_threshold =
-            bsp_assignment_threshold.saturating_sub(T::AssignmentThresholdAsymptote::get());
+    /// Compute the threshold for a BSP to succeed.
+    ///
+    /// Succeeding this threshold is required for the BSP to be eligible to volunteer for a storage request.
+    /// The threshold is computed based on the global reputation weight and the BSP's reputation weight, giving
+    /// an advantage to BSPs with higher reputation weights.
+    ///
+    /// The formalized formulas are documented in the [README](https://github.com/Moonsong-Labs/storage-hub/blob/main/pallets/file-system/README.md#volunteering-succeeding-threshold-checks).
+    pub fn compute_threshold_to_succeed(
+        bsp_id: &ProviderIdFor<T>,
+        requested_at: BlockNumberFor<T>,
+    ) -> Result<(T::ThresholdType, T::ThresholdType), DispatchError> {
+        let maximum_threshold = MaximumThreshold::<T>::get();
 
-        bsp_assignment_threshold = base_threshold
-            .checked_div(&T::AssignmentThresholdDecayFactor::get())
-            .ok_or(Error::<T>::ThresholdArithmeticError)?
-            .saturating_add(T::AssignmentThresholdAsymptote::get());
+        let global_weight =
+            T::ThresholdType::from(T::Providers::get_global_bsps_reputation_weight());
 
-        BspsAssignmentThreshold::<T>::put(bsp_assignment_threshold);
+        if global_weight == T::ThresholdType::zero() {
+            return Err(Error::<T>::NoGlobalReputationWeightSet.into());
+        }
 
-        Ok(())
-    }
+        // Global threshold starting point from which all BSPs begin their threshold slope. All BSPs start at this point
+        // with the starting reputation weight.
+        let threshold_global_starting_point = maximum_threshold
+            .checked_div(&global_weight)
+            .unwrap_or(T::ThresholdType::one())
+            .checked_mul(&ReplicationTarget::<T>::get().into()).unwrap_or({
+            log::warn!("Global starting point is beyond MaximumThreshold. Setting it to half of the MaximumThreshold.");
+            maximum_threshold / 2.into()
```

Comment:
I'd still use `saturated_div(2)` for the last operation. And probably document why we do operations in this order.

Reply:
Adding something like `checked_div` conveys that this can have a result of `0`, which is the only condition for which it returns `None`. Which is impossible in our case since we would never set `maximum_threshold` to be 2 or less.

I will add docs.

Reply:
I understand and agree. The insistence on still using `checked_div` is to avoid at all costs panics in the runtime code. Even if it is impossible.

Reply:
Ok will do.